e2e/rdctl.e2e.spec.ts

```diff
@@ -255,6 +255,92 @@ test.describe('Command server', () => {
   });
 
   test.describe('rdctl', () => {
+    test.describe('config-file and parameters', () => {
+      test.describe("when the config-file doesn't exist", () => {
+        let parameters: string[];
+        const configFilePath = path.join(paths.appHome, 'rd-engine.json');
+        const backupPath = path.join(paths.appHome, 'rd-engine.json.bak');
+        const mvCommand = os.platform() === 'win32' ? 'ren' : 'mv';
+
+        test.beforeAll(async() => {
+          const dataRaw = await fs.promises.readFile(configFilePath, 'utf-8');
+
+          serverState = JSON.parse(dataRaw);
+          parameters = [`--password=${ serverState.password }`,
+            `--port=${ serverState.port }`,
+            `--user=${ serverState.user }`,
+          ];
+          try {
+            await spawnFile(mvCommand, [configFilePath, backupPath]);
+          } catch (err) {
+            console.log(`Error trying to ${ mvCommand } ${ configFilePath } ${ backupPath }: `, err);
+            expect(err).toBeUndefined();
+          }
+        });
+        test.afterAll(async() => {
+          try {
+            await spawnFile(mvCommand, [backupPath, configFilePath]);
+          } catch (err) {
+            console.log(`Error trying to ${ mvCommand } ${ backupPath } ${ configFilePath }: `, err);
+            expect(err).toBeUndefined();
+          }
+        });
+
+        test('it complains with no parameters,', async() => {
+          const { stdout, stderr, error } = await rdctl(['list-settings']);
+
+          expect(error).toBeDefined();
+          expect(stderr).toContain(`Error: open ${ configFilePath }: no such file or directory`);
+          expect(stderr).not.toContain('Usage:');
+          expect(stdout).toEqual('');
+        });
+
+        test('it works with all parameters,', async() => {
+          const { stdout, stderr, error } = await rdctl(parameters.concat(['list-settings']));
+
+          expect(error).toBeUndefined();
+          expect(stderr).toEqual('');
+          expect(stdout).toMatch(/"kubernetes":/);
+          const settings = JSON.parse(stdout);
+
+          expect(['version', 'kubernetes', 'portForwarding', 'images', 'telemetry', 'updater', 'debug', 'pathManagementStrategy']).toMatchObject(Object.keys(settings));
```

Comment:
Is this backwards? You're asserting that this is a _superset_ of the actual properties, I believe.
```suggestion
          expect(settings).toMatchObject({
            version: expect.anything(),
            kubernetes: expect.anything(),
            …
          });
```
(If you take the suggested for, you can also optionally use `.toEqual` instead of `.toMatchObject` to ensure no extra properties.)

Reply:
It wasn't backwards -- I was deliberately verifying that there weren't other fields in the payload. This way sets the required fields, so I'll go with that.